d2/src/main/java/com/linkedin/d2/balancer/simple/SimpleLoadBalancer.java

```diff
@@ -939,6 +938,7 @@ private void handleTimeoutFromGetClusterAndUriProperties(Throwable e, String clu
         getClusterAndUriPropertiesFromCache(clusterName, clusterAndUriPropertiesCallback);
     if (pair != null)
     {
+      _log.info("getClusterAndUriProperties for {} timed out, used cached value instead.", clusterName);
       clusterAndUriPropertiesCallback.onSuccess(pair);
     }
     else
```

Comment:
The callback already seems to be invoked by `die` inside `getClusterAndUriPropertiesFromCache`, so looks like it'll be called twice. Perhaps not a good idea to pass the callback and not call it for both cases anyway.

Reply:
Hi, @shivamgupta1 As u could find in  `getClusterAndUriPropertiesFromCache` , the callback is only called when it could not get info from the cache.
The we need to call it from outside, when it is successful.

Reply:
The case being referred to here is the `else` one below (Github doesn't allow me to select it while commenting): `clusterAndUriPropertiesCallback.onError`

Reply:
Got u. Yes. We should not call it twice, as the `die` will trigger the `onError`.